### PR TITLE
mercurial: update to 4.9.1

### PR DESCRIPTION
**Pull request by ra1nb0w**
#### Description

- bump version to 4.9.1

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 10.14.4 18E226
Xcode 10.2 10E125

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by ra1nb0w** [comment]
It is a minor release but we need to respect: "don't forget to update dependents for mercurial".
I don't find update for the following package:
- py-hgevolve                     
- py-hgevolve-devel               
- py-hgexperimental                               
- py-hgsubversion                 
- py-hgsubversion-devel           
- py-mercurial_extension_utils    
- py-mercurial_keyring            
- tortoisehg 

Now I push an update for py-hggit

**Comment by mojca** [comment]
Tortoisehg was usually broken when mercurial was updated alone, but if this is just a minor bugfix, this might not be an issue. I cannot test now, but if tortoisehg launches for you after the update, it's probably fine.

**Comment by mojca** [comment]
@ra1nb0w: since you seem to be in mass-updating-mode, would you be willing to take a look at https://github.com/macports/macports-base/pull/120?

**Comment by ra1nb0w** [comment]
I updated py-hggit to commit that support mercurial 4.9 but I am not sure if I can move the version to 20190402 and in the future come back to 0.8.13 or a semantic version.

**Comment by ra1nb0w** [comment]
> @ra1nb0w: since you seem to be in mass-updating-mode, would you be willing to take a look at [macports/macports-base#120](https://github.com/macports/macports-base/pull/120)?

Oh. Macports partecipate to GSoC, I need to check out ;-)
Thanks for the link because during these updates I was thinking about something similar to alleviate the simple task of version bump.

File: python/py-hggit/Portfile [hunk 1]
```diff
@@ -6,10 +6,10 @@ PortGroup               bitbucket 1.0
 
 epoch                   20130201
 
-bitbucket.setup         durin42 hg-git 8d00fde45adbc6c3c0ccab8e362b5f5c36c171e6
+bitbucket.setup         durin42 hg-git c8fea7a921634bda3d4e68e2c7a0d73b7447dea4
 name                    py-hggit
-version                 0.8.12
-revision                1
+version                 20190402
```
Comment (mojca) [inline python/py-hggit/Portfile]: To answer your question: if you want to go back to 0.8.13, you need to increase epoch.

Comment (ra1nb0w) [reply python/py-hggit/Portfile]: ok. I don't like to set a fake 0.8.13-alpha or something similar so increment epoch seems the only way

**Comment by ra1nb0w** [comment]
> Tortoisehg was usually broken when mercurial was updated alone, but if this is just a minor bugfix, this might not be an issue. I cannot test now, but if tortoisehg launches for you after the update, it's probably fine.

it starts correctly.

**Comment by mojca** [comment]
Then leave tortoise as it is. Regarding hggit: maybe we should ask upstream to tag a new release.

**Comment by ra1nb0w** [comment]
> Then leave tortoise as it is. Regarding hggit: maybe we should ask upstream to tag a new release.

Nice idea. Now I try

File: python/py-hggit/Portfile (hunk 1 above)
Comment (mojca) [reply python/py-hggit/Portfile]: Anyway, this port already has the epoch set, so it won't be any different if you just increase it.
Still, version `20190402` is not official either.

**Comment by mojca** [comment]
Short note: 4.9.1 seems to be a bugfix release. I have now been bitten by a pretty serious bug in mercurial 4.9.0 (but only on Windows, making repositories with external subrepos useless) and it's a pain because they don't release a new version of Tortoise for Windows with that bugfix.

**Comment by ra1nb0w** [comment]
> Then leave tortoise as it is. Regarding hggit: maybe we should ask upstream to tag a new release.

The developer said that he has another issue  before new release (no ETA at the moment).

**Comment by mojca** [comment]
OK, so what version does py-hggit itself report if you run it (or check the setup or whatever file where the version is defined)? This PR is good to be merged, except that I would also like to avoid artificial version numbers.

**Comment by ra1nb0w** [comment]
The version of `py-hggit` is defined on `hggit/__init__.py` with line `__version__ = '0.8.12'`.
He bumps the number when he tags the new version so at the moment is the release one. Seems there is no progress on the ticket that block the new release ([gitignore doesn't work](https://bitbucket.org/durin42/hg-git/issues/273/gitignore-not-consulted-with-mercurial-49)). At this point we can maintain the same version number and bump revision or set the date of the commit and then, when release exists, set the new version and bump epoch. what do you think?

**Comment by mojca** [comment]
I would usually use the version number `0.8.12-git-20190402` or something similar in such cases. This makes it clear that it's approximately version 0.8.12, but with commits from git, and you know from when.

**Comment by ra1nb0w** [comment]
changed

**Comment by mojca** [comment]
Thanks.

**Comment by ra1nb0w** [comment]
thank you

**Pull request merged**